filter.go

```diff
@@ -224,55 +270,187 @@ func (p *Policy) Dump(out io.Writer) error {
 	return nil
 }
 
-// Assemble assembles the policy into a list of BPF instructions. If the group
-// contains any unknown syscalls or invalid actions an error will be returned.
-func (g *SyscallGroup) Assemble() ([]bpf.Instruction, error) {
-	if len(g.Names) == 0 {
-		return nil, nil
+// SyscallWithConditions consists of a syscall number and optional conditions.
+//
+// The conditions are applied to the arguments of the syscall.
+// So, conditions consist of a list of up to six argument conditions.
+// This filter matches if all argument conditions match for any Conditions.
+type SyscallWithConditions struct {
+	Num        uint32
+	Conditions []ArgumentConditions
+}
+
+// getSyscall searches the syscall in the list.
+// Do not use a map to keep the ordering, as specified by the user.
+func getSyscall(syscalls []SyscallWithConditions, syscall uint32) *SyscallWithConditions {
+	for i := 0; i < len(syscalls); i++ {
+		s := &syscalls[i]
+		if s.Num == syscall {
+			return s
```

Comment:
```suggestion
	for _, s := range syscalls {
		if s.Num == syscall {
			return &s
```

Reply:
Unfortunately this is not possible because the variable `s` of `range` is a copy of the element in the slice and not the element itself. So taking a reference will not work, I added a comment.

Reply:
You are absolutely correct. Thank you. We can use an index range though; `for i := range syscalls {` with what you have that follows.